truffle/contracts/Splitter.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity 0.5.10;
+
+import './Stoppable.sol';
+
+contract Splitter is Stoppable {
+    
+    struct Account {
+        uint    pendingWithdrawals;
+    }
+    
+    mapping(address => Account) accountStorage;
+
+    event LogEthSent(address sender, uint value, address receiver1, address receiver2);
+    event LogWithdrawed(address sender, uint amount);
+    
+    constructor() public payable {}
+    
+    function splitEth(address receiver1, address receiver2) public _onlyIfRunning payable returns(bool) {
+        require(msg.value <= msg.sender.balance, 'The sender balance shoud be higher the ether\'s value');
+
+        accountStorage[receiver1].pendingWithdrawals += msg.value / 2;
+        accountStorage[receiver2].pendingWithdrawals += msg.value / 2;
+
+        emit LogEthSent(msg.sender, msg.value, receiver1, receiver2);
+        
+        return true;
+    }
+    
+    function getBalance() public view returns(uint256) {
+        return address(this).balance;    
+    }
+    
+    function withdraw() public returns(bool) {
+        uint amount = accountStorage[msg.sender].pendingWithdrawals;
+        
+        accountStorage[msg.sender].pendingWithdrawals = 0;
+        msg.sender.transfer(amount);
+        
+        emit LogWithdrawed(msg.sender, amount);
```

Comment:
While the security risk is debatable (I think it's real), it is always a good thing to treat the event as a state change and put it in front of the `.transfer()`. In testing, you will likely find it seems to make no difference. In practice, the coin toss goes to in front. One reason that is easy to demonstrate ... what if msg.sender is a contract and it also emits an event? When reviewing the transaction log, things will seem out of order if everyone follows the current style. 

This would be hard to interpret because the results precede the causes:
Your Dead! => Bang
Fire => Aim => Ready

If the event is in front, then things make more sense:
bang! => you're dead.
ready => aim => fire.